collector/collector.go

```diff
@@ -154,14 +181,34 @@ func (c *collector) Stop() {
 	}
 
 	c.svc.Shutdown()
+
+	shutdownCompleteChan := make(chan struct{})
+
+	go func() {
+		select {
+		case <-ctx.Done():
+			// Cancel the start context if the stop context is cancelled.
+			// Ideally, we'd be able to pass a context into shutdown, but the OTEL collector
```

Comment:
Is that this something we should try to contribute in the future?

Reply:
Yeah, I'll look into writing an issue. Seems like it could be useful upstream.